src/epd7in5_v2/mod.rs

```diff
@@ -223,8 +228,16 @@ where
         self.interface.cmd_with_data(spi, command, data)
     }
 
-    fn wait_until_idle(&mut self) {
-        self.interface.wait_until_idle(IS_BUSY_LOW)
+    fn wait_until_idle(&mut self, spi: &mut SPI, delay: &mut DELAY) -> Result<(), SPI::Error> {
+        self.interface.wait_until_idle(
+            IS_BUSY_LOW,
+            Some((spi, delay, |i, s, d| {
+                i.cmd(s, Command::GET_STATUS)?;
+                d.delay_ms(20);
+                Ok(())
+            })),
+        )?;
+        Ok(())
```

Comment:
See the command in the interface. But wouldn't it be easier to just handle everything here already?

Reply:
No idea why I made it this complicated.
Should be as easy as
```
while self.interface.is_busy(IS_BUSY_LOW) {
    self.interface.cmd(Command::GET_STATUS)?;
    delay.delay_ms(20);
}
Ok(())
```